src/PlatformSwitch/hooks.ts

```diff
@@ -0,0 +1,57 @@
+import { redirect } from '@graasp/sdk';
+
+/** Enumeration of available platforms */
+export enum Platform {
+  Builder = 'Builder',
+  Player = 'Player',
+  Library = 'Library',
+  Analytics = 'Analytics',
+}
+
+/** Maps each Platform to a URL generator function */
+export type HostsMapper = Partial<Record<Platform, (itemId: string) => string>>;
+
+/**
+ * Generates a default hosts mapper given a record of platform to hostname
+ * Only the origin part of the given hostname will be used!
+ *
+ * The default mapping is:
+ *  BUILDER_HOST/items/<itemId>
+ *  PLAYER_HOST/<itemId>
+ *  LIBRARY_HOST/<itemId>
+ *
```

Comment:
For analytics it is also `<host>/items/<id>`

Reply:
Aren't the analytics currently hosted under Builder?